packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,449 @@
 
 part of firebase_ml_vision;
 
+/// Detector for performing optical character recognition(OCR) on an input image.
+///
+/// A barcode detector is created via getVisionBarcodeDetector() in [FirebaseVision]:
+///
+/// ```dart
+/// BarcodeDetector barcodeDetector = FirebaseVision.instance.barcodeDetector();
+/// ```
 class BarcodeDetector extends FirebaseVisionDetector {
-  BarcodeDetector._(BarcodeDetectorOptions options);
+  BarcodeDetector._();
 
+  /// Closes the barcode detector and release its model resources.
   @override
   Future<void> close() async {
-    // TODO: implement close
+    return FirebaseVision.channel.invokeMethod('BarcodeDetector#close');
   }
 
+  /// Detects barcode in the input image.
+  ///
+  /// The OCR is performed asynchronously.
   @override
   Future<void> detectInImage(FirebaseVisionImage visionImage) async {
-    // TODO: implement detectInImage
+    final List<dynamic> reply = await FirebaseVision.channel.invokeMethod(
+      'BarcodeDetector#detectInImage',
+      visionImage.imageFile.path,
+    );
+
+    final List<TextBlock> blocks = <TextBlock>[];
```

Comment:
I believe you want `List<VisionBarcode> barcodes <VisionBarcode>[];` right?

Reply:
Woops... It's my fault.